src/backend/backend.hpp

```diff
@@ -118,6 +118,33 @@ namespace argo {
 		 */
 		void release();
 
+		/**
+		 * @brief causes a node to self-invalidate the specified pages from its cache,
+		 *        and thus getting any updated values on subsequent accesses
+		 * @param addr	the start of the memory region to self-invalidate
+		 * @param size  the size of the memory region to self-invalidate. A size equal
+		 * to zero results in that no pages are selectively acquired.
+		 * @note  Selective coherence operations do not uphold the data consistency
+		 * semantics of regular coherence operations and should be used with this
+		 * in mind. Specifically, write ordering with respect to writes outside of
+		 * the selective coherence region is not upheld.
+		 */
+		void selective_acquire(void* addr, size_t size);
```

Comment:
cppreference states that implicit conversions occur (and are allowed):
> A prvalue pointer to any (optionally cv-qualified) object type T can be converted to a prvalue pointer to (identically cv-qualified) void. The resulting pointer represents the same location in memory as the original pointer value. If the original pointer is a null pointer value, the result is a null pointer value of the destination type.

And that the following qualifies as a `prvalue`
> &a, the built-in address-of expression;

I therefore assume that the conversion from:
`argo::backend::selective_acquire(T& object, sizeof(T));`
to:
`argo::backend::selective_acquire(void* addr, size_t size);`
is both implicit and allowed.

The selective coherence operations make no assumption about the type of the address they are passed, and treat it solely as a memory address. 

The reason for this choice is originally just an old implementation relic, but I can at the same time not see a simple way (without multiple explicit instantiations) of providing separate templated definitions of `argo::backend::selective_acquire/release` for each backend. The definition of the templated function must be available at compile time, as the linker will otherwise not find a matching definition for each `T` without explicit instantiations if I understand things correctly. Am I missing a straightforward way to accomplish this?

Reply:
This is now changed according to the original suggestion:
`selective_acquire(T* addr, size_t size);` and
`selective_release(T* addr, size_t size);` respectively.